content/git/concepts/gitignore/gitignore.md

```diff
@@ -0,0 +1,96 @@
+* System files such as **thumbs.db** on Windows or **.DS_Store** on macOS
+
+## Creating a `.gitignore` file
+
+We can create a **.gitignore** file easily using a terminal editor like *nano* or *emacs*, or just using a File explorer. Don’t forget the `.` before the filename!
+
+**.gitignore** is usually placed in the root directory of the repository. The filenames inside a **.gitignore** file can be written relative to the location of the **.gitignore** file. For example, we could add the line
+```
+src/main.js
+
+```
+to ignore the file **main.js** under the **src/** directory.
+
+![alt text](https://static-assets.codecademy.com/Courses/learn-git-github/gitignore/create-gitignore.png "A screenshot of a Linux terminal showing the creation of a .gitignore file using the command "touch .gitignore". Next, the commands "ls" and "ls -a" are used to show that since .gitignore is a hidden file, the "-a" flag is necessary")
+
+**Note:** since **.gitignore** is a hidden file, we will need to add the `-a` flag to `ls` to see it.
+
+## Ignore a directory with .gitignore
+
+Sometimes we want to ignore entire directories or specify certain files in a directory. Common directories to leave out of a Git repository are **node_modules** or **logs** folder. We can ignore an entire directory by simply adding its name to **.gitignore**:
+```
+node_modules/
+```
+This will ignore the **node_modules** directory, and all subdirectories and files inside them. The forward slash `/` specifies that we are ignoring the directory.
+
+## `.gitignore` Patterns
+
+We can take advantage of patterns to match multiple filenames. These help us handle special cases such as ignoring specific file types or ignoring all but one file inside a directory. Some examples of things that make up patterns are:
+
+* Wildcard * to match 0 or more characters except for /. For example, adding `*.html` to **.gitignore** would ignore all files ending with the `.html` extension. `example*` would match any file starting with example such as `example.txt` or `exampleHtmlFile.html`.
+* Negation `!` as a prefix to negate any file that would otherwise be ignored. For example,
+```
+index*
+!public/index.css
+```
+will ignore all files starting with *index* except for *src/index.css*. But, we cannot negate a file inside an ignored directory.
+
+* Square brackets `[]` can be used to match a single character from a set of characters or a range of characters. Note that the range can be alphabetical: `[a-z]` or `[A-Z]`, numeric `[0-9]`, or a set of characters. If we added `index.[a-i]*` with both the square bracket and wildcard to **.gitignore**, we would ignore *index.css* and *index.html* but not *index.js*, since “j” is outside of the `[a-i]` range.
+* Double asterisk `**` is used to match 0 or more directories. If we had a temp folder inside all of the folders in the root directory and we only wanted to match files with the `.log` extension, we could use the pattern `**/temp/*.log`.
```

Comment:
Try to separate the explanation and usage details of different patterns.

Add a list of patterns possible:

```
- `*` matches zero or more characters.
- `**` matches files in all subdirectories.
- `!` negates an ignore rule.
- `/` specifies a directory.
```

And then you can show usage in different examples like this - 

```
# Ignore all `.txt` files
*.txt

# Ignore all files in the `temp` directory
temp/**

# Do not ignore `important.txt` inside `temp`
!temp/important.txt
```

I suggest not to mix both at the same place, as it will make the text long, difficult to follow and understand.